unlock-app/src/components/interface/locks/List/elements/LockList.tsx

```diff
@@ -21,14 +26,31 @@ const LocksByNetwork = ({ network, owner }: LocksByNetworkProps) => {
   if (locks?.length === 0) return null
 
   return (
-    <div className="flex flex-col gap-4">
-      <h2 className="text-lg font-bold text-brand-ui-primary">{networkName}</h2>
-      <div className="flex flex-col gap-6">
-        {locks?.map((lock: Lock, index: number) => (
-          <LockCard key={index} lock={lock} network={network} />
-        ))}
-        {loading && <LockCardPlaceholder />}
-      </div>
+    <div className="w-full">
+      <Disclosure defaultOpen>
+        {({ open }) => (
+          <div className="flex flex-col gap-2">
+            <Disclosure.Button className="flex items-center justify-between w-full outline-none ring-0">
+              <h2 className="text-lg font-bold text-brand-ui-primary">
+                {networkName}
+              </h2>
+              {open ? (
+                <UpIcon className="fill-brand-ui-primary" size={24} />
+              ) : (
+                <DownIcon className="fill-brand-ui-primary" size={24} />
+              )}
+            </Disclosure.Button>
+            <Disclosure.Panel>
+              <div className="flex flex-col gap-6">
+                {locks?.map((lock: Lock, index: number) => (
+                  <LockCard key={index} lock={lock} network={network} />
+                ))}
+                {loading && <LockCardPlaceholder />}
+              </div>
+            </Disclosure.Panel>
+          </div>
+        )}
+      </Disclosure>
     </div>
```

Comment:
Think this part is repeated a few times. Let's ask chiali on a universal collapse and use it everywhere?